src/core/tokenizer.ts

```diff
@@ -39,29 +42,79 @@ const FIXED_TOKENS = [
   "<!--",
   "-->",
 ];
-const REPEATABLE_SYMBOLS_REGEX = REPEATABLE_SYMBOLS.map(escapeRegExp)
-  .map((s) => `${s}+`)
-  .join("|");
-const FIXED_TOKENS_REGEX = FIXED_TOKENS.map(escapeRegExp).join("|");
+
 const IDENTIFIERS_REGEX = "[\\p{L}_0-9]+";
 const SINGLE_SYMBOLS_REGEX = "[^\\s\\w]";
 const NUMBERS_REGEX = "(?<=[^.\\d]|^)\\d+\\.\\d+(?=[^.\\d]|$)"; // (not-dot/digit digits dot digits not-dot/digit)
 
-const REGEX = [
-  FIXED_TOKENS_REGEX,
-  NUMBERS_REGEX,
-  IDENTIFIERS_REGEX,
-  REPEATABLE_SYMBOLS_REGEX,
-  SINGLE_SYMBOLS_REGEX,
-].join("|");
+const defaultLanguageTokenizerSetting: DefaultLanguageTokenizerSettings = {
+  fixedTokens: FIXED_TOKENS,
+  repeatableSymbols: REPEATABLE_SYMBOLS,
+  identifiersRegex: IDENTIFIERS_REGEX,
+  numbersRegex: NUMBERS_REGEX,
+  singleSymbolsRegex: SINGLE_SYMBOLS_REGEX,
+};
+
+export const TOKEN_MATCHER = generateTokenMatcher();
+
+function generateTokenMatcher(languageId?: SupportedLanguageId): RegExp {
+  let settings = {};
+  if (languageId && languageSettings[languageId]) {
+    settings = languageSettings[languageId] ?? {};
```

Comment:
why do you need `??` if you're already checking it's non-null in guard?

Reply:
I think this might have something to do with `Partial`. I'm getting a type error here that `settings` could be `undefined` which is why I added null-ish coalesce.

Reply:
It's prob because you access the array twice: once in the guard and once here

Reply:
ty